lib/virt_autotest/utils.pm

```diff
@@ -42,39 +42,58 @@ sub trim {
     return $text;
 }
 
-sub restart_libvirtd {
-    if (is_sle('<12')) {
-        assert_script_run('rclibvirtd restart', 180);
-    }
-    elsif (is_alp) {
-        my $_libvirtd_pid = script_output(q@ps -ef |grep [l]ibvirtd | gawk '{print $2;}'@);
-        my $_libvirtd_cmd = script_output("ps -o command $_libvirtd_pid | tail -1");
-        assert_script_run("kill -9 $_libvirtd_pid");
-        assert_script_run("$_libvirtd_cmd");
-    }
-    else {
-        systemctl("restart libvirtd", timeout => 180);
-    }
-    record_info("Libvirtd has been restarted!");
-    save_screenshot;
-}
-
 #return 1 if test is expected to run on XEN hypervisor
 sub is_xen_host {
     return get_var("XEN") || check_var("SYSTEM_ROLE", "xen") || check_var("HOST_HYPERVISOR", "xen") || check_var("REGRESSION", "xen-hypervisor");
 }
 
+#Setup modular libvrit daemons as the default in SLE15SP6, refer to PED-4804 to get more details
+sub is_modular_libvirtd {
+    return 0 if is_sle(">=15-sp6");
+}
+
+# Usage: check_modular_libvirt_daemons([daemon1_name daemon2_name ...]). For example:
+# to specify daemons which will be checked: check_modular_libvirt_daemons(qemu storage ...)
+# to check all required modular daemons without any daemons passed
+sub check_modular_libvirt_daemons {
+    my @daemons;
+
+    if (@_ == 0) {
+        @daemons = qw(network nodedev nwfilter secret storage proxy log lock);
+        if (is_xen_host) {
+            push @daemons, 'xen';
+        } else {
+            push @daemons, 'qemu';
+        }
+    } else {
+        @daemons = @_;
+    }
+
+    foreach my $daemon (@daemons) {
+        systemctl("status virt${daemon}d.service");
+        if ($daemon == 'lock') {
+            systemctl("status virt${daemon}d\{,-admin\}.socket");
+        } else {
+            systemctl("status virt${daemon}d\{,-ro,-admin\}.socket");
+        }
+    }
+
+    record_info("Modular Libvirt daemons checked", join(' ', @daemons));
+    save_screenshot;
+}
+
 # Usage: restart_modular_libvirt_daemons([daemon1_name daemon2_name ...]). For example:
 # to specify daemons which will be restarted: restart_modular_libvirt_daemons(virtqemud virtstoraged ...)
 # to restart all modular daemons without any daemons passed
 sub restart_modular_libvirt_daemons {
     my @daemons;
 
     if (@_ == 0) {
+        @daemons = qw(network nodedev nwfilter secret storage proxy log lock);
```

Comment:
since you change the daemons name from `virtnetworkd` to `network`, the usage of this funtions is changed too, fg, it was used as `restart_modular_libvirt_daemons('virtnetworkd')` to `restart_modular_libvirt_daemons('network')`.   The name becomes shorter but is a bit harder to be used. but I really like the 'push' below.

Reply:
oh, I overlooked that there are two seperate parts to handle the daemons. then it is ok. ignore my comment at here.